river_dl/preproc_utils.py

```diff
@@ -391,7 +391,7 @@ def reduce_training_data_random(
     df = ds.to_dataframe()
     idx = pd.IndexSlice
     df_trn = df.loc[idx[train_start_date:train_end_date, :], :]
-    if segs:
+    if np.any(segs):
```

Comment:
Could you use is not None instead? I think that's the preferred method for comparison statements.
```suggestion
    if segs is not None:
```

Reply:
Sure. I just switched them all to `if segs is not None:`

I also wondered about other files. I did a quick search of `postproc_utils.py` (assuming the metric calculation might be another likely place) and didn't find any, but I didn't check any other files.